### PR TITLE
Loadable extensions

### PR DESCRIPTION
**Pull request by phiresky**
Trying to fix #631 to allow creation of loadable (.so / .dll) extensions for sqlite.

I rebased it to current master.

I can try to fix issues that pop up, but I'm not the original author of most of these changes and not really knowledgable on FFI in general sadly.

### CODE REVIEW

**Comment by phiresky** [comment]
The reason the tests are failing is that the "dummy-extension" module is a workspace member, and `--workspace` causes dummy-extension to be compiled together with libsqlite3-sys, which apparently activates extension mode for all tests which means they fail with `sqlite3_api is null`.

`cargo test -p libsqlite3-sys -p rusqlite` works, while

`cargo test -p libsqlite3-sys -p rusqlite -p dummy-extension` doesn't.

**Comment by thomcc** [comment]
Neither of those is great. I guess removing it from the workspace and adding it to exclude would work.

File: libsqlite3-sys/src/lib.rs [hunk 1]
```diff
@@ -50,6 +50,26 @@ pub enum Limit {
     SQLITE_LIMIT_WORKER_THREADS = 11,
 }
 
+#[cfg(feature = "loadable_extension_embedded")]
+// bindings were built with loadable_extension_embedded:
+// define sqlite3_api as an extern since this code will be embedded
+// within a loadable extension that defines and exports this itself
+extern "C" {
+    #[no_mangle]
+    pub static mut sqlite3_api: *mut sqlite3_api_routines;
```
Comment (thomcc) [inline libsqlite3-sys/src/lib.rs:59]: This should probably be an AtomicPtr, and have a Release store and Acquire loads. Otherwise it's a data race.

**Review by gwenn — COMMENTED** [review]
I will try to fix warnings and build issues.

**Comment by jrandall** [comment]
See a note in #910 regarding this PR not adhering to the conditions of the license under which code contributed in #532 was provided: https://github.com/rusqlite/rusqlite/pull/910#issuecomment-791485983

The issue here appears to have arisen out of a commit inherited from #631 in which the "Genomics plc" copyright notice was removed and not (as far as I can tell) recreated elsewhere: https://github.com/rusqlite/rusqlite/pull/868/commits/68bea40e806df992d11ff8d12480738fc34bb611

I am sure this was not done maliciously, but these contributions are not owned by me and I only have permission from the owner (my employer) to contribute it under the terms of the license, including the specific copyright notice.

Please stop distributing this code without including the required copyright notice. The easiest way to do that might be to revert (or rebase out of existence) the commit 68bea40e806df992d11ff8d12480738fc34bb611

**Comment by gwenn** [comment]
> See a note in #910 regarding this PR not adhering to the conditions of the license under which code contributed in #532 was provided: [#910 (comment)](https://github.com/rusqlite/rusqlite/pull/910#issuecomment-791485983)
> 
> The issue here appears to have arisen out of a commit inherited from #631 in which the "Genomics plc" copyright notice was removed and not (as far as I can tell) recreated elsewhere: [68bea40](https://github.com/rusqlite/rusqlite/commit/68bea40e806df992d11ff8d12480738fc34bb611)
> 
> I am sure this was not done maliciously, but these contributions are not owned by me and I only have permission from the owner (my employer) to contribute it under the terms of the license, including the specific copyright notice.
> 
> Please stop distributing this code without including the required copyright notice. The easiest way to do that might be to revert (or rebase out of existence) the commit [68bea40](https://github.com/rusqlite/rusqlite/commit/68bea40e806df992d11ff8d12480738fc34bb611)

See https://github.com/rusqlite/rusqlite/blob/68bea40e806df992d11ff8d12480738fc34bb611/LICENSE
> Copyright (c) 2019 Genomics plc <info@genomicsplc.com>
So [68bea40] explicitly removed `Genomics plc` from `build.rs` but kept it untouched in global LICENCE file (there is no file with copyright header in this repo)...
And https://github.com/rusqlite/rusqlite/pull/631#discussion_r404001452
`Genomics plc` was removed from global LICENSE file when master branch was merged.

**Comment by phiresky** [comment]
Superseded by #910 (mostly due to the license discussion?)

File: libsqlite3-sys/src/lib.rs (hunk 1 above)
Comment (jrandall) [reply libsqlite3-sys/src/lib.rs:59]: Discussion moved to https://github.com/rusqlite/rusqlite/pull/910/files#r765305470

**Pull request closed without merging**